### PR TITLE
Rebase geth 1.7.0

### PR DESCRIPTION
**Pull request by adambabik**
Upgrades to geth 1.7.0.

## Tasks

- [x] upgrade vendor with our go-ethereum 1.7.0 fork,
- [x] upgrade our cluster to geth 1.7.0 and update lists of bootnodes,
- [x] run tests and apply fixes if needed,
- [ ] update CHT as geth 1.7.0 has them newer,
- [x] regenerate `static/bindata.go`.

### CODE REVIEW

**Comment by adambabik** [comment]
This PR is blocked by missing bootnodes. We currently have only one running geth 1.7.0 but it looks like it's not enough. I couldn't sync the blockchain using it.

**Comment by divan** [comment]
Geth 1.7.1 is out: https://github.com/ethereum/go-ethereum/releases/tag/v1.7.1

**Comment by tiabc** [comment]
https://github.com/status-im/status-go/issues/381

**Comment by adambabik** [comment]
I have run all tests that I could and it looks good! I updated a Todo list in this PR. After all points are done we can merge it.

File: geth/node/node.go [hunk 1]
```diff
@@ -65,25 +65,25 @@ func MakeNode(config *params.NodeConfig) (*node.Node, error) {
 		stackConfig.P2P.PrivateKey = pk
 	}
 
+	// TODO(divan): FIXME: this is rude workaround for #294 issue
```
Comment (adambabik) [inline geth/node/node.go]: This should be removed before merged.

File: geth/account/accounts.go [hunk 2]
```diff
@@ -144,24 +144,34 @@ func (m *Manager) RecoverAccount(password, mnemonic string) (address, pubKey str
 	return address, pubKey, nil
 }
 
+type accountKeyFile struct {
+	Address string `json:"address"`
+}
```
Comment (tiabc) [inline geth/account/accounts.go]: Can we make this struct inline? We're only using it in a single place for a single reason.

**Comment by tiabc** [comment]
In progress of updating the cluster.

**Comment by tiabc** [comment]
Updated the cluster to geth1.7.1

**Comment by tiabc** [comment]
Regarding the tasks specified in the description:
1. Bootnodes list needn't be updated except for 1 new node.
2. CHT update isn't urgent any more as we're using remote nodes by default and I consider this is out of scope of this issue.

Please, get the nodes list back not deleting the new one: `enode://a1ef9ba5550d5fac27f7cbd4e8d20a643ad75596f307c91cd6e7f85b548b8a6bf215cca436d6ee436d6135f9fe51398f8dd4c0bd6c6a0c332ccb41880f33ec12@51.15.218.125:30303`, re-generate `bindata.go` and let me know so that I merged it.

**Comment by adambabik** [comment]
@tiabc it's done.

**Comment by tiabc** [comment]
Can you, please, ensure all tests pass before we merge this? For example, try this:
`build/env.sh go test -v ./geth/api/... -testify.m TestCallRPCSendTransaction`

I get the same result as in ci: https://travis-ci.org/status-im/status-go/builds/285739848#L730

**Comment by adambabik** [comment]
It's because the blockchain was not synced. In Travis after rerun:
```
--- PASS: TestCallRPCSendTransaction (33.26s)
```

**Pull request merged**